src/marqo/tensor_search/models/settings_object.py

```diff
@@ -0,0 +1,127 @@
+
+settings_schema = {
+    "$schema": "https://json-schema.org/draft/2019-09/schema",
+    "type": "object",
+    "required": [
+        "index_defaults",
+        "number_of_shards"
```

Comment:
just confirming that this change happens after the autofill, otherwise it would be a breaking change for people who create `index_settings` with just `index_defaults` or just `number_of_shards`

Reply:
Yep validation happens after _autofill: https://github.com/marqo-ai/marqo/blob/validate_settings.py/src/marqo/tensor_search/tensor_search.py#L89